android/java/org/chromium/chrome/browser/vpn/billing/InAppPurchaseWrapper.java

```diff
@@ -87,15 +86,13 @@ public boolean isMonthlySubscription(String productId) {
                 || productId.equals(RELEASE_MONTHLY_SUBSCRIPTION);
     }
 
-    public boolean isBillingClientReady() {
-        return mBillingClient != null && mBillingClient.isReady();
-    }
-
     public void startBillingServiceConnection(
             Context context, MutableLiveData<Boolean> billingClientConnectionState) {
-        if (!BraveVpnUtils.isBraveVpnFeatureEnable()) {
+        Log.e(TAG, " after startBillingServiceConnection");
+        if (!BraveVpnUtils.isVpnFeatureSupported(context)) {
             return;
         }
+        Log.e(TAG, " after startBillingServiceConnection");
         mBillingClient = BillingClient.newBuilder(context)
```

Comment:
I think we should call `endConnection` if `mBillingClient != null` before creating a new object.